frontend/packages/console-app/src/components/nodes/NodesPage.tsx

```diff
@@ -40,60 +57,117 @@ const NodeTableHeader = () => {
-      </TableData>
-    </TableRow>
-  );
+const mapStateToProps = ({ UI }) => ({
+  metrics: UI.getIn(['metrics', 'node']),
+});
+
+type NodesRowMapFromStateProps = {
+  metrics: NodeMetrics;
 };
+
+const NodesTableRow = connect<NodesRowMapFromStateProps, null, NodesTableRowProps>(mapStateToProps)(
+  ({ obj: node, index, key, style, metrics }: NodesTableRowProps & NodesRowMapFromStateProps) => {
+    const nodeName = getName(node);
+    const nodeUID = getUID(node);
+    const usedMem = metrics?.usedMemory?.[nodeName];
+    const totalMem = metrics?.totalMemory?.[nodeName];
+    const memory =
+      Number.isFinite(usedMem) && Number.isFinite(totalMem)
+        ? `${humanizeBinaryBytes(usedMem).string} / ${humanizeBinaryBytes(totalMem).string}`
+        : '-';
+    const cores = metrics?.cpu?.[nodeName];
+    const usedStrg = metrics?.usedStorage?.[nodeName];
+    const totalStrg = metrics?.totalStorage?.[nodeName];
+    const storage =
+      Number.isFinite(usedStrg) && Number.isFinite(totalStrg)
+        ? `${humanizeBinaryBytes(usedStrg).string} / ${humanizeBinaryBytes(totalStrg).string}`
+        : '-';
+    const pods = metrics?.pods?.[nodeName] ?? '-';
+    return (
+      <TableRow id={nodeUID} index={index} trKey={key} style={style}>
+        <TableData className={tableColumnClasses[0]}>
+          <ResourceLink kind={referenceForModel(NodeModel)} name={nodeName} title={nodeUID} />
+        </TableData>
+        <TableData className={tableColumnClasses[1]}>
+          <NodeStatus node={node} />
+        </TableData>
+        <TableData className={tableColumnClasses[2]}>
+          <NodeRoles node={node} />
+        </TableData>
+        <TableData className={tableColumnClasses[3]}>{pods}</TableData>
+        <TableData className={tableColumnClasses[4]}>{memory}</TableData>
+        <TableData className={tableColumnClasses[5]}>
+          {cores ? `${formatCores(cores)} cores` : '-'}
+        </TableData>
+        <TableData className={tableColumnClasses[6]}>{storage}</TableData>
+        <TableData className={tableColumnClasses[7]}>
+          {fromNow(node.metadata.creationTimestamp)}
+        </TableData>
+        <TableData className={tableColumnClasses[8]}>
+          <ResourceKebab
+            actions={menuActions}
+            kind={referenceForModel(NodeModel)}
+            resource={node}
+          />
+        </TableData>
+      </TableRow>
+    );
+  },
+);
 NodesTableRow.displayName = 'NodesTableRow';
+
 type NodesTableRowProps = {
   obj: NodeKind;
   index: number;
   key?: string;
   style: object;
 };
 
-const NodesTable: React.FC<NodesTableProps> = (props) => (
-  <Table {...props} aria-label="Nodes" Header={NodeTableHeader} Row={NodesTableRow} virtualize />
-);
+const NodesTable: React.FC<NodesTableProps> = React.memo((props) => {
+  const row = React.useCallback(
+    (rowProps: NodesTableRowProps) => <NodesTableRow {...rowProps} />,
+    [],
```

Comment:
No dependencies to `useCallback`?

Reply:
Yes not required since we are not injecting anything to row externally in the `useCallback` block.